### PR TITLE
Feature/revert precompiled

### PR DESCRIPTION
**Pull request by krlosMata**
This PR does the following:
- revert when calling a precompiled that is not supported
- add zk-counter check `PADDING_PG_LIMIT`
- assure 32 bit range for certain free inputs

### CODE REVIEW

**Review by ignasirv — APPROVED** [review]
💯

File: main/utils.zkasm [hunk 1]
```diff
@@ -438,8 +438,16 @@ computeGasSendCall:
...
+
+    ; C = [c7, c6, ..., c0]
+    ; JMPN instruction assures c0 is within the range [0, 2**32 - 1]
+    ${GAS >> 6} => C        :JMPN(failAssert)
     ${GAS & 0x3f} => D
+
+    ; since D is assured to be less than 0x40
+    ; it is enforced that [c7, c6, ..., c1] are 0 since there is no value multiplied by 64
+    ; that equals the field
+    ; Since e0 is assured to be less than 32 bits, c0 * 64 + d0 could not overflow the field
```
Comment (chriseth) [inline main/utils.zkasm:450]: `e0` -> `c0`

File: main/utils.zkasm [hunk 2]
```diff
@@ -438,8 +438,16 @@ computeGasSendCall:
...
     ; compute all_but_one_64th gas
     GAS => A
-    ${GAS >> 6} => C
+
+    ; C = [c7, c6, ..., c0]
+    ; JMPN instruction assures c0 is within the range [0, 2**32 - 1]
+    ${GAS >> 6} => C        :JMPN(failAssert)
     ${GAS & 0x3f} => D
+
+    ; since D is assured to be less than 0x40
```
[the remark this answers is not in the document]
Comment (chriseth) [reply main/utils.zkasm:447]: Similar comments on the other cases below.

[the remark this answers is not in the document]
Comment (krlosMata) [reply main/utils.zkasm:447]: `D` is set to `A` and later on, it is assured that is less than `0x40` with:
```
D => A
0x40 => B
    $                       :LT,JMPNC(failAssert)
```

[the remark this answers is not in the document]
Comment (chriseth) [reply main/utils.zkasm:447]: Ah, thanks!

**Pull request merged**